### PR TITLE
Allows some snacks to be dunked in liquids

### PR DESCRIPTION
**Pull request by Dingo-Dongler**
## About The Pull Request
This PR adds an afterattack method to snacks to allow snacks with the dunkable trait to be dunked in glasses or open drinks to transfer reagents. Currently this applies to donuts, bread/slices and misc food like cookies. If you feel something should be dunkable probably add it in here.

## Why It's Good For The Game
It simply adds an extra thing to do at lunchtime to get a little cozier. 

## Changelog
:cl:
add: Some foods can now be dunked in drinks. Dip your breakfast donut in some robust coffee today!
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
Gonna poison all the pigs donuts.

**Comment by wesoda25** [comment]
good first pr

**Comment by Shadowflame909** [comment]
hey captain want a box of donuts

I promise the last one in the box doesn't have the roundstart gib toxin in it

File: code/modules/food_and_drinks/food/snacks.dm [hunk 1]
```diff
@@ -330,6 +332,23 @@ All foods are distributed among various categories. Use common sense.
 					M.emote("me", 1, "[sattisfaction_text]")
 				qdel(src)
 
+/obj/item/reagent_containers/food/snacks/afterattack(obj/item/reagent_containers/M, mob/user, proximity)
+	. = ..()
+	if(!dunkable)
```
Comment (nemvar) [inline code/modules/food_and_drinks/food/snacks.dm]: You need to check for proximity. Afterattack can be called from a distance.

File: code/modules/food_and_drinks/food/snacks.dm [hunk 2]
```diff
@@ -330,6 +332,23 @@ All foods are distributed among various categories. Use common sense.
...
+		return
+	if(istype(M, /obj/item/reagent_containers/glass) || istype(M, /obj/item/reagent_containers/food/drinks))	//you can dunk dunkable snacks into beakers or drinks
+		if(istype(M, /obj/item/reagent_containers/food/drinks/soda_cans)) //to prevent dunkage through soda can lids
+			if(!M.spillable)
```
Comment (nemvar) [inline code/modules/food_and_drinks/food/snacks.dm]: Just check for is_drainable() in general.

File: code/modules/food_and_drinks/food/snacks.dm [hunk 3]
```diff
@@ -330,6 +332,23 @@ All foods are distributed among various categories. Use common sense.
...
+				to_chat(user, "<spane class='warning'>[M] is unopened!</span>")
```
Comment (nemvar) [inline code/modules/food_and_drinks/food/snacks.dm]: spane class

File: code/modules/food_and_drinks/food/snacks.dm [hunk 4]
```diff
@@ -330,6 +332,23 @@ All foods are distributed among various categories. Use common sense.
...
+				return
+		if(M.reagents.trans_to(src, dunk_amount, transfered_by = user))	//if reagents were transfered, show the message
+			to_chat(user, "<span class='notice'>You dunk \the [src] into \the [M].</span>")
+		else			//if not, either the beaker was empty, or the snack was full
```
Comment (nemvar) [inline code/modules/food_and_drinks/food/snacks.dm]: You can just return and remove one layer of nesting.

**Comment by bgobandit** [comment]
fries should be dunkable

**Pull request merged**